home/components/ProjectView.tsx

```diff
@@ -0,0 +1,399 @@
+        <ProjectContents app={data.app.byId} />
+      </ScrollView>
+    );
+  }
+
+  React.useLayoutEffect(() => {
+    if (data?.app.byId) {
+      const fullName = data?.app.byId.fullName;
+      navigation.setOptions({
+        title: fullName,
+        headerRight: () => <ShareProjectButton fullName={fullName} />,
+      });
+    }
+  }, [navigation, data?.app.byId]);
+
+  return <View style={{ flex: 1 }}>{contents}</View>;
+}
+
+function ProjectContents({ app }: { app: ProjectDataProject }) {
+  const isLatestLegacyPublishDeprecated = React.useMemo<boolean>(() => {
+    const majorVersionString = app.sdkVersion.split('.').shift();
+    if (majorVersionString) {
+      const majorVersion = parseInt(majorVersionString, 10);
+      return majorVersion < Environment.lowestSupportedSdkVersion;
+    }
+    return false;
+  }, [app.sdkVersion]);
+
+  return (
+    <>
+      <ProjectHeader app={app} />
+      {isLatestLegacyPublishDeprecated && <ExpoSDKOutdated sdkVersion={app.sdkVersion} />}
+      <StartButton
+        title="Open Project"
+        disabled={isLatestLegacyPublishDeprecated}
+        onPress={() => {
+          Linking.openURL(UrlUtils.normalizeUrl(app.fullName!));
+        }}
+      />
+      <EASUpdateLaunchSection app={app} />
+    </>
+  );
+}
+
+function EASUpdateLaunchSection({ app }: { app: ProjectDataProject }) {
+  const hasEASUpdates = app.updateChannels.some(channel =>
+    channel.updateBranches.some(branch => branch.updates.length > 0)
+  );
+
+  if (!hasEASUpdates) {
+    return null;
+  }
+
+  const onPress = (updateChannel: ProjectUpdateChannel) => {
+    const updateBranches = updateChannel.updateBranches;
+    if (updateBranches.length === 0) {
+      Alert.alert(`No update branches on channel: ${updateChannel.name}`);
+      return;
+    }
+    const firstUpdateBranch = updateBranches[0];
+    const updates = firstUpdateBranch.updates;
+    if (updates.length === 0) {
+      Alert.alert(
+        `No updates on channel "${updateChannel.name}" and branch "${firstUpdateBranch.name}"`
+      );
+      return;
+    }
+    const platformUpdates = updates.filter(update => update.platform === Platform.OS);
+    if (platformUpdates.length === 0) {
+      Alert.alert(
+        `No updates for platform "${Platform.OS}" on channel "${updateChannel.name}" and branch "${firstUpdateBranch.name}"`
+      );
+      return;
+    }
+
+    const sdkUpdates = platformUpdates.filter(update =>
+      isSDKVersionRuntimeVersion(update.runtimeVersion)
+    );
+    if (sdkUpdates.length === 0) {
+      Alert.alert(
+        `No Expo SDK runtime updates for platform "${Platform.OS}" on channel "${updateChannel.name}" and branch "${firstUpdateBranch.name}"`
+      );
+      return;
+    }
+
+    const update = sdkUpdates[0];
+
+    Linking.openURL(
+      `exps://updates.expo.dev/${app.id}?runtime-version=${update.runtimeVersion}&channel-name=${updateChannel.name}`
```

Comment:
We're considering changing this URL's domain. Once we go to production we need to keep old URLs working, but we might want to have GraphQL return the URL instead of assuming the domain here.

Reply:
Yeah, the issue is that the only url we have for getting a manifest is by channel and runtime version (which lives on an update), so it's unclear where this field would live.
- `Update.url` would mean that most urls would not actually reference that update in particular, so this is a no go. We could add a new router for launching a particular update.
- `Branch.latestUpdateURL` would need to fetch the latest update for that branch and then construct a URL with that update's runtime version.
- `Channel.latestUpdateURL` would need to fetch the latest update on that channel and then construct a URL with that update's runtime verison. This probably makes the most sense for now.
- Some root GraphQL field like `updateBaseURL`. This seems a bit hacky.

Overall I think what this will look like in the near future will change significantly as I think we'll want the ability to launch individual updates and therefore fetch their manifests.